src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -58,7 +62,8 @@ const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubm
   }, [txsWithDetails, multiSendTxs])
 
   const onExecute = async () => {
-    if (!onboard || !multiSendTxData || !multiSendContract || !txsWithDetails) return
+    if (!onboard || !multiSendTxData || !multiSendContract || !txsWithDetails || !chain || !wallet || !maxFeePerGas)
```

Comment:
This is the equivalent of a loading flag. `maxFee` is [always estimated](https://github.com/safe-global/safe-wallet-web/blob/underpriced-ledger/src/hooks/useGasPrice.ts#L82), it only returns `undefined` whilst `getGasPrice` is being called.

Reply:
Ideally we should refactor `useGasPrice` to return a loading flag.

Reply:
Looking at the code it seems that it can be undefined if the fetch doesn't resolve or throws although I am not 100% sure on the value if it throws. I suggest we write more tests for `useGasPrice` and cover those cases where the fetch doesn't behave as expected.